src/libs/cropOrRotateImage/index.ts

```diff
@@ -74,33 +66,29 @@ function cropCanvas(canvas, options) {
     result.height = height;
 
     const context = result.getContext('2d');
-    context.drawImage(canvas, originX, originY, width, height, 0, 0, width, height);
+    context?.drawImage(canvas, originX, originY, width, height, 0, 0, width, height);
 
     return result;
 }
 
-/**
- * @param {Object} canvas
- * @param {Object} options
- * @returns {Promise<File>}
- */
-function convertCanvasToFile(canvas, options = {}) {
+function convertCanvasToFile(canvas: HTMLCanvasElement, options: CropOrRotateImageOptions): Promise<FileWithUri> {
     return new Promise((resolve) => {
         canvas.toBlob((blob) => {
+            if (!blob) {
+                return;
+            }
             const file = new File([blob], options.name || 'fileName.jpeg', {type: options.type || 'image/jpeg'});
-            file.uri = URL.createObjectURL(file);
-            resolve(file);
+            const fileWithUri = Object.assign(file, {uri: URL.createObjectURL(file)});
+
+            resolve(fileWithUri);
```

Comment:
```ts
            const file = new File([blob], options.name || 'fileName.jpeg', {type: options.type || 'image/jpeg'}) as FileWithUri;
            file.uri = URL.createObjectURL(file);
            resolve(file);
```

In this way you don't need to use `Object.assign`

Reply:
Actually no, I need to do that or I don't know better way because `uri` dont originally exists on type File. When I will use Object.assign `fileWithUri` will have type of 
`File & {uri: string}` which is equal to `FileWithUri` which I expect to be returned from that function

Reply:
Fixed @fabioh8010